.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,83 @@
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: cd/Configure AWS
+        uses: aws-actions/configure-aws-credentials@07c2f971bac433df982ccc261983ae443861db49 # v1-node16
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.PR_BUILDS_BUCKET_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.PR_BUILDS_BUCKET_AWS_SECRET_ACCESS_KEY }}
+      - name: cd/Download artifacts
+        uses: dawidd6/action-download-artifact@0c49384d39ceb023b8040f480a25596fd6cf441b # v2.26.0
+        with:
+          workflow_conclusion: success
+          name: server-dist-artifact
+      # Get Branch name from calling workflow
+      # Search for the string "pull" and replace it with "PR" in branch-name
+      - name: cd/Get branch name
+        run: echo "BRANCH_NAME=$(echo ${{ github.event.workflow_run.head_branch }} | sed 's/^pull\//PR-/g')" >> $GITHUB_ENV
+      - name: cd/Upload artifacts to S3
+        run: |
+          aws s3 cp ${{github.workspace}}/ s3://pr-builds.mattermost.com/$REPO_NAME/$BRANCH_NAME/ --acl public-read --cache-control "no-cache" --recursive --no-progress
+          aws s3 cp ${{github.workspace}}/ s3://pr-builds.mattermost.com/$REPO_NAME/commit/${{ github.sha }}/ --acl public-read --cache-control "no-cache" --recursive --no-progress
+  build-docker:
+    name: cd/Build and push docker image
+    needs: upload-s3
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    runs-on: ubuntu-22.04
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+    - name: cd/Login to Docker Hub
+      uses: docker/login-action@3da7dc6e2b31f99ef2cb9fb4c50fb0971e0d0139 # v2.1.0
+      with:
+        username: ${{ secrets.DOCKERHUB_DEV_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_DEV_TOKEN }}
+    - name: cd/Download artifacts
+      uses: dawidd6/action-download-artifact@0c49384d39ceb023b8040f480a25596fd6cf441b # v2.26.0
+      with:
+        workflow_conclusion: success
+        name: server-build-artifact
+        path: server/build/
+    - name: cd/Setup Docker Buildx
+      uses: docker/setup-buildx-action@11e8a2e2910826a92412015c515187a2d6750279 # v2.4
+    - name: cd/Docker build and push
+      env:
+        DOCKER_CLI_EXPERIMENTAL: enabled
+      run: |
+        export TAG=$(echo "${{ github.event.pull_request.head.sha || github.sha }}" | cut -c1-7)
+        cd server/build
+        export DOCKER_CLI_EXPERIMENTAL=enabled
+        export MM_PACKAGE=https://pr-builds.mattermost.com/$REPO_NAME/commit/${{ github.sha }}/mattermost-team-linux-amd64.tar.gz
+        docker buildx build --push --build-arg MM_PACKAGE=$MM_PACKAGE -t mattermostdevelopment/mm-te-test:${TAG} .
+  sentry:
+    name: Send build info to sentry
```

Comment:
Shouldn't we wait for the builds ?

Reply:
When is the sentry file generated ?

Reply:
Sentry needs: 

```
    needs:
      - test-postgres-binary
      - test-postgres-normal
      - test-mysql
      - build-mattermost-server
```

These are all covered with: 
```
on:
  workflow_run:
    workflows: ["Mattermost Build"]
    types:
      - completed
```

Reply:
These tests generate a file I think . Is this file available on the next step ?

Reply:
🤷 I have no context on this. 
Is the file that you refer to, available to the previous state of the pipeline? 

https://github.com/mattermost/mattermost-server/blob/master/.github/workflows/ci.yml#L292-L308

Reply:
We should publish release sha only if this gets merged into master

Reply:
@phoinixgrr This needs to run only when branch is master ?

Reply:
This runs on every PR 
```
    if: >
      github.event.workflow_run.event == 'pull_request' &&
      github.event.workflow_run.conclusion == 'success'
```

Should it run only on Master? 
cc: @isacikgoz